File: Android/RecyclerView/app/src/main/java/com/example/listacontato/adapter/ContatoAdapter.java
package com.example.listacontato.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.listacontato.R;
import com.example.listacontato.model.Contatos;

import java.util.List;

public class ContatoAdapter extends RecyclerView.Adapter<ContatoAdapter.ViewHolder> {

    private List<Contatos> listaContatos;

    public ContatoAdapter(List<Contatos> listaContatos) {
        this.listaContatos = listaContatos;
    }

    @NonNull
    @Override
    //Infla o layout do item e retorna uma nova instancia do ViewHolder
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.layout_lista_item_contato, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        //Criar uma nova instancia
        Contatos novoContato = listaContatos.get(i);

        //adiciona o valor da instancia para ser atribuido no componente
        viewHolder.bind(novoContato);
    }

    //Retorna o tamanho da lista
    @Override
    public int getItemCount() {
        return listaContatos.size();
    }




    // CLASSE ViewHolder
    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView nomeContato;
        TextView telefonContato;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            nomeContato = itemView.findViewById(R.id.txtNome);
            telefonContato = itemView.findViewById(R.id.txtTelefone);
        }

        public void bind(Contatos contato) {
            nomeContato.setText(contato.getNome());
            telefonContato.setText(contato.getTelefone());
        }
    }
}

File: Android/agendaKotlin/app/src/main/java/com/example/agendakotlin/view/adapter/ContatoAdapter.kt
package com.example.agendakotlin.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.agendakotlin.R
import com.example.agendakotlin.model.Contato

class ContatoAdapter (var contatos : List<Contato>, val clickItem : (contato: Contato) -> Unit  ) : RecyclerView.Adapter<ContatoAdapter.ViewHolder>(){

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        var view = LayoutInflater.from(parent.context).inflate(R.layout.layout_item_lista_contato, parent, false)
        return ViewHolder(view)

    }

    override fun getItemCount(): Int {
        return contatos.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        var contato : Contato = contatos[position]
        holder.bind(contato)

        holder.itemView.setOnClickListener {
            clickItem(contato)
        }
    }


    class ViewHolder (itemView : View) : RecyclerView.ViewHolder(itemView){

        var txtNome = itemView.findViewById<TextView>(R.id.txtNome)
        var txtTelefone = itemView.findViewById<TextView>(R.id.txtTelefone)

        fun bind(contato: Contato){


            txtNome.text = contato.nome
            txtTelefone.text = contato.telefone

        }

    }
}